app/controllers/memes_controller.rb

```diff
@@ -0,0 +1,51 @@
+class MemesController < ApplicationController
+  before_action :authenticate_user!
+
+  def show
+    @meme = Meme.find(params[:id])
+  end
+
+  def new
+    @category = Category.find(params[:category_id])
+    @user = current_user.id
+    @meme = Meme.new
+  end
+
+  def create
+    @category = Category.find(params[:category_id])
+
+    if user_signed_in?
+      @user = current_user
+      @meme = Meme.new(meme_params)
+      @meme.category = @category
+      @meme.user = @user
+
+      if @meme.save
+        @category.memes << @meme
+        flash[:notice] = "Meme successfully added"
+        redirect_to category_path(@category)
+      else
+        flash[:notice] = @meme.errors.full_messages.to_sentence
+        @memes = @category.memes
+        render '/categories/show'
+      end
+
+    else
+      flash[:notice] = "You must be logged in to do that"
+      redirect_to category_path(@category)
+    end
+  end
+
+  def edit
+  end
+
+  def destroy
+  end
```

Comment:
I'll allow this this time, but in the future, only create the methods if you are going to use them in THIS PR!